### DISCUSSION TITLE
Testing with unity

### DISCUSSION THREAD

**Discussion by taliesin** [dzurikmiroslav/esp32-evse#69] [Ideas]
I've just read the article about [unit testing in ESP32](https://docs.espressif.com/projects/esp-idf/en/stable/esp32/api-guides/unit-tests.html#tests-for-different-targets]).
It defines a specific file structure for testing:

- Unit tests are located in the test sub-directory of a component
- The test subdirectory should contain a minimal component CMakeLists.txt, like:
```
idf_component_register(SRC_DIRS "."
                       INCLUDE_DIRS "."
                       REQUIRES unity)
```
- In the tools/unit-test-app configure and build the test application, so there is only one test application, that includes all tests.

The current unity-test branch does not follow this structure. Why is that so?

**Comment by dzurikmiroslav** [comment]
Yes you have the right, unit test should be placed in the component directory.
I didn't know how to solve it better, that in the `evse` test I need access to the mock of `peripherals`. I didn't want to put the `peripherals` mock in the `evse` component.
To generate the test main app, I used the VSCode extension _Unit Test: Build and flash unit test app for testing_, it created the `unity-app` directory.

It is not final yet, you can propose a better solution.

**Comment by dzurikmiroslav** [comment]
So in the end I was inspired by unit tests in ESP-IDF components. With the use of the Unity Fixtures.
All tests are in the `test_app` directory, while test group is only for the `evse` and `script` module. The `peripheral` module is mocked.
